src/java/org/apache/cassandra/tools/NodeProbe.java

```diff
@@ -1140,6 +1146,11 @@ public void truncateHints()
         hsProxy.deleteAllHints();
     }
 
+    public List<PendingHintsInfo> listPendingHints()
```

Comment:
I saw you made `PendingHintsInfo` serializable but I'm not sure how all this stuff works if you connected via JMX directly. Could you please confirm that the current implementation works through both nodetool and JMX.

My general understanding was that we prefer returning basic objects (lists, maps, strings, etc) from MBeans and NodeProbe instead of custom types, but I'm not sure. I'll let @ekaterinadimitrova2 to comment here.

Reply:
Agreed! Changed in https://github.com/apache/cassandra/pull/1232/commits/8787a817ca6b646cf7be10d70f65e0beb5be69bf.